client/src/procedures/primitives.rs

```diff
@@ -519,19 +532,33 @@ fn x25519_secret_key(raw: Ref<u8>) -> Result<x25519::SecretKey, crypto::Error> {
 
 fn ed25519_secret_key(raw: Ref<u8>) -> Result<ed25519::SecretKey, crypto::Error> {
     let mut raw = (*raw).to_vec();
-    if raw.len() < ed25519::SECRET_KEY_LENGTH {
+    if raw.len() < ed25519::SecretKey::LENGTH {
         let e = crypto::Error::BufferSize {
             has: raw.len(),
-            needs: ed25519::SECRET_KEY_LENGTH,
+            needs: ed25519::SecretKey::LENGTH,
             name: "data buffer",
         };
         return Err(e);
     }
-    raw.truncate(ed25519::SECRET_KEY_LENGTH);
-    let mut bs = [0; ed25519::SECRET_KEY_LENGTH];
+    raw.truncate(ed25519::SecretKey::LENGTH);
+    let mut bs = [0; ed25519::SecretKey::LENGTH];
     bs.copy_from_slice(&raw);
 
-    Ok(ed25519::SecretKey::from_bytes(bs))
+    Ok(ed25519::SecretKey::from_bytes(&bs))
+}
+
+fn secp256k1_ecdsa_secret_key(raw: Ref<u8>) -> Result<secp256k1_ecdsa::SecretKey, crypto::Error> {
+    let raw_slice: &[u8] = &raw;
+    if raw_slice.len() < secp256k1_ecdsa::SecretKey::LENGTH {
+        let e = crypto::Error::BufferSize {
+            has: raw_slice.len(),
+            needs: secp256k1_ecdsa::SecretKey::LENGTH,
+            name: "data buffer",
```

Comment:
```suggestion
            name: "secp256k1 data buffer",
```

Nit: Consider providing a bit more info in the error for debugging.

Reply:
Done